sympy/sets/conditionset.py

```diff
@@ -133,19 +164,23 @@ def __new__(cls, sym, condition, base_set=S.UniversalSet):
 
         condition = as_Boolean(condition)
 
-        if isinstance(sym, Tuple):  # unsolved eqns syntax
-            return Basic.__new__(cls, sym, condition, base_set)
-
-        if not isinstance(base_set, Set):
-            raise TypeError('expecting set for base_set')
+        if condition is S.true:
+            return base_set
 
         if condition is S.false:
             return S.EmptySet
-        elif condition is S.true:
-            return base_set
+
         if isinstance(base_set, EmptySet):
             return base_set
 
+        # no simple answers, so now check syms
+        for i in flat:
+            if not getattr(i, '_diff_wrt', False):
+                raise ValueError('`%s` is not symbol-like' % i)
```

Comment:
This seems like an odd way of checking

Reply:
I am open to other suggestions...we don't have a `is_symbol_like` property.